bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/AbstractPresentableCalendar.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icalendar.internal.logic;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A calendar which provides the interface to the calendar implementation for
+ * the binding, encapsulating the implementation of the real calendar.
+ *
+ * @author Michael Wodniok - Initial contribution
+ * @author Andrew Fiddian-Green - Methods getJustBegunEvents() & getJustEndedEvents()
+ *
+ */
+@NonNullByDefault
+public abstract class AbstractPresentableCalendar {
+
+    /**
+     * Creates an implementing Instance of AbstractPresentableCalendar.
+     *
+     * @param calendarStream A Stream containing the iCal data.
+     * @return The instance.
+     * @throws IOException When something while reading stream fails.
+     * @throws CalendarException When something while parsing fails.
+     */
+    public static AbstractPresentableCalendar create(InputStream calendarStream) throws IOException, CalendarException {
+        return new BiweeklyPresentableCalendar(calendarStream);
```

Comment:
Do you expect to add other calendars other than BiweeklyPresentableCalendar?

Reply:
I've already replaced the whole implementation once. While replacing, i've created this way of semi-loose binding to the calendar implementation.